src/components/PopupWithForm.js

```diff
@@ -26,8 +40,29 @@ export default class PopupWithForm extends Popup {
       e.preventDefault();
       const inputValues = this._getInputValues();
       this._handleFormSubmit(inputValues);
+      this._popupForm.reset();
       this.close();
     });
     super.setEventListeners();
   }
+
+  setSubmitAction(action) {
+    this._handleFormSubmit = action;
+  }
+
+  renderLoading(isLoading) {
+    if (isLoading) {
+      this._modalButton.textContent = "Saving...";
+    } else {
+      this._modalButton.textContent = "Save";
+    }
+  }
+
+  renderCreating(isLoading) {
+    if (isLoading) {
+      this._modalButton.textContent = "Creating...";
+    } else {
+      this._modalButton.textContent = "Create";
```

Comment:
There should **not** be any code in universal classes that doesn’t suit every popup.